viewing_party/party.py

```diff
@@ -1,23 +1,186 @@
 # ------------- WAVE 1 --------------------
 
+from tests.test_constants import GENRE_1, MOVIE_TITLE_1, RATING_1
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+
+
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie
+    else: 
+        return None
+
+    
+def add_to_watched(user_data, movie):
+    user_data = {"watched": []}
+    
+    movie = {
+        "title": MOVIE_TITLE_1,
+        "genre": GENRE_1,
+        "rating": RATING_1
+    }
+
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist": []}
+    movie = {
+    "title": MOVIE_TITLE_1,
+    "genre": GENRE_1,
+    "rating": RATING_1
+    }
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+
+    
+    for key, value in user_data.items():
+        if key == "watchlist":

```

Comment:
Every user represented by user_data has a key called "watchlist" that has a value that is a list (either empty or contains dictionaries representing movies they want to watch), therefore we do not need to iterate through the key and values of user_data and check if we have a key called "watchlist".

We can directly iterate over user_data["watchlist"]. 

When we directly iterate over the watchlist, we're iterating over dictionaries that represent movies and each dictionary has a key called "title", as outlined by the README.

Additionally, if we use a for-in loop then we can directly access the movie (dictionary) without using indices. 

So your refactored function could look like: 
```python
def watch_movie(user_data, title):
    for movie in user_data["watchlist"]:
        if movie["title"] == title:
            user_data["watched"].append(movie)
            user_data["watchlist"].remove(movie)

    return user_data